x-pack/plugins/observability_solution/dataset_quality/server/routes/data_streams/get_data_stream_details/index.ts

```diff
@@ -7,28 +7,163 @@
-import { DataStreamDetails } from '../../../../common/api_types';
+import {
+  findInventoryFields,
+  InventoryItemType,
+  inventoryModels,
+} from '@kbn/metrics-data-access-plugin/common';
+import { rangeQuery } from '@kbn/observability-plugin/server';
+
+import { MAX_HOSTS_METRIC_VALUE } from '../../../../common/constants';
+import { _IGNORED } from '../../../../common/es_fields';
+import { DataStreamDetails, DataStreamSettings } from '../../../../common/api_types';
+import { createDatasetQualityESClient } from '../../../utils';
 import { dataStreamService } from '../../../services';
 
-export async function getDataStreamDetails(args: {
+export async function getDataStreamSettings({
+  esClient,
+  dataStream,
+}: {
+  esClient: ElasticsearchClient;
+  dataStream: string;
+}): Promise<DataStreamSettings> {
+  throwIfInvalidDataStreamParams(dataStream);
+
+  const createdOn = await getDataStreamCreatedOn(esClient, dataStream);
+
+  return {
+    createdOn,
+  };
+}
+
+export async function getDataStreamDetails({
+  esClient,
+  dataStream,
+  start,
+  end,
+  sizeStatsAvailable = true,
+}: {
   esClient: ElasticsearchClient;
   dataStream: string;
+  start: number;
+  end: number;
+  sizeStatsAvailable?: boolean; // Only Needed to determine whether `_stats` endpoint is available https://github.com/elastic/kibana/issues/178954
 }): Promise<DataStreamDetails> {
-  const { esClient, dataStream } = args;
+  throwIfInvalidDataStreamParams(dataStream);
 
-  if (!dataStream?.trim()) {
-    throw badRequest(`Data Stream name cannot be empty. Received value "${dataStream}"`);
+  try {
+    const dataStreamSummaryStats = await getDataStreamSummaryStats(
+      esClient,
+      dataStream,
+      start,
+      end
+    );
+
+    const whenSizeStatsNotAvailable = NaN; // This will indicate size cannot be calculated
+    const avgDocSizeInBytes = sizeStatsAvailable
+      ? dataStreamSummaryStats.docsCount > 0
+        ? await getAvgDocSizeInBytes(esClient, dataStream)
+        : 0
+      : whenSizeStatsNotAvailable;
+    const sizeBytes = Math.ceil(avgDocSizeInBytes * dataStreamSummaryStats.docsCount);
+
+    return {
+      ...dataStreamSummaryStats,
+      sizeBytes,
+    };
+  } catch (e) {
+    // Respond with empty object if data stream does not exist
+    if (e.statusCode === 404) {
+      return {};
+    }
+    throw e;
   }
+}
 
+async function getDataStreamCreatedOn(esClient: ElasticsearchClient, dataStream: string) {
   const indexSettings = await dataStreamService.getDataSteamIndexSettings(esClient, dataStream);
 
   const indexesList = Object.values(indexSettings);
 
-  const indexCreationDate = indexesList
+  return indexesList
     .map((index) => Number(index.settings?.index?.creation_date))
     .sort((a, b) => a - b)[0];
+}
+
+type TermAggregation = Record<string, { terms: { field: string; size: number } }>;
+
+const MAX_HOSTS = MAX_HOSTS_METRIC_VALUE + 1; // Adding 1 so that we can show e.g. '50+'
```

Comment:
At the moment we are thinking about `MAX = 50` for aggregating services and hosts, is this enough @ruflin?

Reply:
We'll do this in a follow up PR, at the moment we are just showing a `+` whenever we encounter more than `50`